Replace icon fonts with Gridicons

willgorham opened the pull request:
Closes #16903 

The objective of this PR is to replace icon fonts with Gridicons SVG icons ([repo](https://github.com/Automattic/gridicons) / [gallery](http://automattic.github.io/gridicons/)) and drop IE8 support. 

<hr/>

Lots of replacements here, so I'll add some screenshots as reference to what changed. In general, I followed the recommendations of @jameskoster here: https://github.com/woocommerce/woocommerce/issues/16903#issuecomment-331901122. There were quite a few icons not listed in that comment though. 

### Notes/considerations:

Gridicons provides a php helper function `get_gridicon()` that will return the SVG markup (located in `includes/libraries/gridicons.php`). I also dropped the raw .svg files into `assets/images/icons/gridicons/` so they can be referenced by path as well.

_I'm thinking `get_gridicon` could be added to the PHPCS exceptions for escaping content?_ Otherwise it will throw a violation when this is used.

In general I tried to put all SVGs inline instead of in pseudo-elements (`::before/::after`) primarily because as far as I know it's impossible to style the fill color of SVGs in pseudo-elements. Also relative url paths in .css files are messy, and this keeps concerns separated. The downside of this is it will break the styling of some extensions relying on core styles being in pseudo-elements, and same for themes in the few front-end-facing cases. I can switch back to some pseudo-element icons if you'd rather keep things as backward-compatible as possible.

If this is approved, I guess the next step would be dropping IE8 support in the Gruntfile?

<hr/>

The icons I *_didn't_* change are:
- The main Woo icon in the admin menu per https://github.com/woocommerce/woocommerce/issues/16903#issuecomment-331901122 (maybe a shopping cart?).
- The product list table types column (pending whatever happens to that). There are Gridicons for simple, virtual, external, and downloadable products. We'd need something for variable and grouped. **Edit: This column was removed entirely**
- The order table actions buttons. I think we could just go with text here? I think I recall a past issue about these being confusing. **Edit: These are now plain text**
- The dashboard widget. I think this is going away as well? If that's not happening soon, I could change these for now.
- The review/rating star, per https://github.com/woocommerce/woocommerce/pull/19445#issuecomment-374211584

Most notable replacements:
- ~~Review star: This will cause styling breakage for themes that define their own star CSS (like Storefront). Also, the dynamic 'add a review' stars are added by JS so are styled in a pseudo-element, which means that you can't change their color and they look different than all of the other stars. I'm not sure what the best solution here would be, or if there needs to be one.~~
- Front-end notices: Could affect custom themes as the icons are now inline and not in ::before. 
- Product/coupon metabox data tabs: Will affect extensions that add their own icons. I added support for declaring a Gridicon when adding a custom tab.
- Table sorting handles: There are a bunch of these with a lot of duplicate styling. Might be worth consolidating things into a mixin here. Some are inline and some are in pseudo elements currently.
- Remove item crosses ('X') on the layered nav / filter widgets: Could affect custom themes due to icons being inline now.

<details>
<summary>Screenshots</summary>

<details>
<summary>Admin menu</summary>
<img width="175" alt="screen shot 2018-03-17 at 5 36 45 pm" src="https://user-images.githubusercontent.com/1883584/37567204-da506c10-2a99-11e8-9e5f-a75be76a69f8.png">
</details>
<details>
<summary>Tooltip</summary>
<img width="575" alt="screen shot 2018-03-17 at 5 38 34 pm" src="https://user-images.githubusercontent.com/1883584/37567209-ef6cbb62-2a99-11e8-88f8-eb376b65731a.png"><br/><img width="357" alt="screen shot 2018-03-17 at 5 39 34 pm" src="https://user-images.githubusercontent.com/1883584/37567210-ef7bab86-2a99-11e8-9cde-7d17db5de436.png"><br/><img width="310" alt="screen shot 2018-03-17 at 5 41 00 pm" src="https://user-images.githubusercontent.com/1883584/37567211-ef889f94-2a99-11e8-927f-e452309f2d95.png">
</details>
<details>
<summary>Blank states</summary>
<img width="587" alt="screen shot 2018-03-17 at 6 29 26 pm" src="https://user-images.githubusercontent.com/1883584/37567185-a340d246-2a99-11e8-9475-03818495c08c.png">
<img width="544" alt="screen shot 2018-03-17 at 5 52 47 pm" src="https://user-images.githubusercontent.com/1883584/37567222-3fba9ada-2a9a-11e8-8990-9be82dcadd17.png">
<img width="511" alt="screen shot 2018-03-17 at 6 01 22 pm" src="https://user-images.githubusercontent.com/1883584/37567226-5050570e-2a9a-11e8-8c75-4a4ebb1acf99.png">
<img width="573" alt="screen shot 2018-03-17 at 6 03 44 pm" src="https://user-images.githubusercontent.com/1883584/37567229-54aa2050-2a9a-11e8-8d5e-3ac3a0660d65.png">
<img width="555" alt="screen shot 2018-03-17 at 6 04 13 pm" src="https://user-images.githubusercontent.com/1883584/37567233-5a06bd92-2a9a-11e8-8935-87df9b24b5d1.png">
</details>
<details>
<summary>Status page check/notice/X</summary>
<img width="1261" alt="screen shot 2018-03-18 at 10 53 33 am" src="https://user-images.githubusercontent.com/1883584/37567252-a2ed14a2-2a9a-11e8-8f80-39488379cad4.png">
<img width="1262" alt="screen shot 2018-03-18 at 10 53 59 am" src="https://user-images.githubusercontent.com/1883584/37567257-afeef0bc-2a9a-11e8-8be4-a7145a369c13.png">
</details>
<details>
<summary>Products list table image header</summary>
This:<br/>
<img width="271" alt="screen shot 2018-03-17 at 5 45 39 pm" src="https://user-images.githubusercontent.com/1883584/37567275-f31a8cca-2a9a-11e8-8bdc-a6d507e76143.png">

Became this:  
<img width="301" alt="screen shot 2018-03-17 at 5 46 08 pm" src="https://user-images.githubusercontent.com/1883584/37567276-f32b46a0-2a9a-11e8-80c9-f44af4321acb.png">
</details>
<details>
<summary>Featured product star</summary>
<img width="251" alt="screen shot 2018-03-17 at 5 49 23 pm" src="https://user-images.githubusercontent.com/1883584/37567296-4aac3aa6-2a9b-11e8-80ca-f21ee51743b0.png">
</details>
<details>
<summary>Product meta box data tabs</summary>
<img width="353" alt="screen shot 2018-03-17 at 5 50 47 pm" src="https://user-images.githubusercontent.com/1883584/37567303-642c89f4-2a9b-11e8-9e9f-0858dc99f0b3.png">
</details>
<details>
<summary>Coupon meta box data tabs</summary>
<img width="351" alt="screen shot 2018-03-18 at 10 59 37 am" src="https://user-images.githubusercontent.com/1883584/37567307-7a66667c-2a9b-11e8-873e-82c593052693.png">
</details>
<details>
<summary>Table sorting handles</summary>
<img width="139" alt="screen shot 2018-03-17 at 6 05 11 pm" src="https://user-images.githubusercontent.com/1883584/37567326-d3f98d68-2a9b-11e8-9f6d-de1df8f76a57.png">
<br/><img width="416" alt="screen shot 2018-03-17 at 6 08 18 pm" src="https://user-images.githubusercontent.com/1883584/37567329-db33c95e-2a9b-11e8-881f-5cedfa9e58ba.png"><br/><img width="981" alt="screen shot 2018-03-17 at 6 09 21 pm" src="https://user-images.githubusercontent.com/1883584/37567335-f6b1a1d8-2a9b-11e8-9e5c-bf0cc56de2c9.png">
</details>
<details>
<summary>Shipping zones globe</summary>
<img width="299" alt="screen shot 2018-03-17 at 6 07 07 pm" src="https://user-images.githubusercontent.com/1883584/37567344-0c9f4388-2a9c-11e8-94a4-c9e944d7d99b.png">
</details>
<details>
<summary>Payment gateway enabled</summary>
<img width="1008" alt="screen shot 2018-03-17 at 6 09 56 pm" src="https://user-images.githubusercontent.com/1883584/37567356-312ca1be-2a9c-11e8-8fbc-28dad7bbb0b4.png">
</details>
<details>
<summary>Emails enabled/disabled/manually sent</summary>
<img width="360" alt="screen shot 2018-03-17 at 6 11 25 pm" src="https://user-images.githubusercontent.com/1883584/37567362-41003aba-2a9c-11e8-85da-1f871c66b853.png">
</details>
<details>
<summary>Email configure cog (removed)</summary>
Could easily go back to a cog icon if desired.
<img width="1250" alt="screen shot 2018-03-17 at 6 11 54 pm" src="https://user-images.githubusercontent.com/1883584/37567376-78ffbbd4-2a9c-11e8-816e-2b3da4536d26.png">
</details>
<details>
<summary>Reports export button</summary>
<img width="185" alt="screen shot 2018-03-17 at 6 13 10 pm" src="https://user-images.githubusercontent.com/1883584/37567384-90427548-2a9c-11e8-831d-07e15d7bdef1.png">
</details>
<details>
<summary>Shipping zones blank state globe</summary>
The Gridicon 'globe' is pretty weak. It looks more like a soccer ball when it's big 😄<br/>
<img width="1193" alt="screen shot 2018-03-17 at 6 15 14 pm" src="https://user-images.githubusercontent.com/1883584/37567391-a87a82ea-2a9c-11e8-84b7-d3289ff46c38.png">
</details>
<details>
<summary>Order edit addresses</summary>
<img width="623" alt="screen shot 2018-03-17 at 5 54 39 pm" src="https://user-images.githubusercontent.com/1883584/37567439-4a60d42e-2a9d-11e8-9134-4c317b2ad73b.png">
</details>
<details>
<summary>Order item edit/delete</summary>
<a href="https://cl.ly/qFFE" target="_blank"><img src="https://d3vv6lp55qjaqc.cloudfront.net/items/2c0g0x0I000l0Z351810/Screen%20Recording%202018-03-17%20at%2005.56%20PM.gif" style="display: block;height: auto;width: 100%;"/></a>
</details>
<details>
<summary>Order shipping/fee/refund line item</summary>
<img width="968" alt="screen shot 2018-03-17 at 5 55 02 pm" src="https://user-images.githubusercontent.com/1883584/37567527-b6fd15b0-2a9e-11e8-9074-8ccbbb890da9.png">
</details>
<details>
<summary>Order item image placeholder (replaced with default placeholder)</summary>
<img width="342" alt="screen shot 2018-03-17 at 5 58 19 pm" src="https://user-images.githubusercontent.com/1883584/37567540-d7d79c7e-2a9e-11e8-9fa8-d6e7519ad0c3.png">
</details>
<details>
<summary>Order action (replaced with text)</summary>
<img width="307" alt="screen shot 2018-03-17 at 5 58 58 pm" src="https://user-images.githubusercontent.com/1883584/37567549-f5887450-2a9e-11e8-95ca-48826333a161.png">
</details>
<details>
<summary>Order preview</summary>
<a href="https://cl.ly/qGq5" target="_blank"><img src="https://d3vv6lp55qjaqc.cloudfront.net/items/1Q1E290W1t3S3B122K2e/Screen%20Recording%202018-03-17%20at%2005.53%20PM.gif" style="display: block;height: auto;width: 100%;"/></a>
</details>
<details>
<summary>Rating star</summary>
<a href="https://cl.ly/qFpM" target="_blank"><img src="https://d3vv6lp55qjaqc.cloudfront.net/items/2b3c0A04000v3r3B0T1G/wc-icons-star.png" style="display: block;height: auto;width: 100%;"/></a><br/><a href="https://cl.ly/qFh0" target="_blank"><img src="https://d3vv6lp55qjaqc.cloudfront.net/items/1E1d0e3t0A2z0o2e2v2E/Image%202018-03-17%20at%205.31.28%20PM.png" style="display: block;height: auto;width: 100%;"/></a>
</details>
<details>
<summary>Product variation image add/delete</summary>
<a href="https://cl.ly/qGg6" target="_blank"><img src="https://d3vv6lp55qjaqc.cloudfront.net/items/2Q1j442G3R0n1a0a3Z3W/Screen%20Recording%202018-03-17%20at%2006.19%20PM.gif" style="display: block;height: auto;width: 100%;"/></a>
</details>
<details>
<summary>AJAX add to cart loading/added</summary>
<a href="https://cl.ly/qFuh" target="_blank"><img src="https://d3vv6lp55qjaqc.cloudfront.net/items/3C05123I0X202X040K3g/Screen%20Recording%202018-03-17%20at%2006.20%20PM.gif" style="display: block;height: auto;width: 100%;"/></a>
</details>
<details>
<summary>Notice/info/error messages</summary>
<img width="773" alt="screen shot 2018-03-17 at 6 23 09 pm" src="https://user-images.githubusercontent.com/1883584/37567624-ee954de8-2a9f-11e8-875f-60368382670a.png"><br/><br/>
<img width="759" alt="screen shot 2018-03-17 at 6 23 42 pm" src="https://user-images.githubusercontent.com/1883584/37567625-eea6686c-2a9f-11e8-8a6b-83f69f3fdc42.png">
<br/><br/><img width="753" alt="screen shot 2018-03-17 at 6 25 48 pm" src="https://user-images.githubusercontent.com/1883584/37567626-eebadd4c-2a9f-11e8-86e1-91dd636b8906.png">
</details>
<details>
<summary>Order coupon remove X</summary>
<img width="142" alt="screen shot 2018-03-18 at 3 52 22 pm" src="https://user-images.githubusercontent.com/1883584/37570314-b1e31d7e-2ac4-11e8-86e8-9ba8a0af7472.png">
</details>
<details>
<summary>My Account email/phone</summary>
<img width="390" alt="screen shot 2018-03-18 at 5 04 38 pm" src="https://user-images.githubusercontent.com/1883584/38126949-c83d5ef8-33c1-11e8-9432-4ad6d3a3729d.png">
</details>
<details>
<summary>Layered nav, rating filter, and active filters widgets 'remove' X</summary>
<img width="365" alt="screen shot 2018-03-29 at 8 25 11 pm" src="https://user-images.githubusercontent.com/1883584/38127028-3d597bd6-33c2-11e8-964f-3faf9c5cd669.png">
</details>
<details>
<summary>Product importer complete check</summary>
<img width="723" alt="screen shot 2018-03-29 at 8 47 24 pm" src="https://user-images.githubusercontent.com/1883584/38127078-7eb0e72c-33c2-11e8-89af-76a3680c3490.png">
</details>
<details>
<summary>Reports - chart widget resize handle</summary>
<img width="250" alt="screen shot 2018-03-29 at 9 03 55 pm" src="https://user-images.githubusercontent.com/1883584/38127095-9a477bfe-33c2-11e8-9de7-cf01bec2d022.png">
</details>
<details>
<summary>Product downloadable files 'remove' X</summary>
<img width="549" alt="screen shot 2018-03-29 at 9 19 56 pm" src="https://user-images.githubusercontent.com/1883584/38127125-b5479c86-33c2-11e8-8c0b-7cf1fb0eff0b.png">
</details>
<details>
<summary>Admin product gallery remove X and placeholder</summary>
<a href="https://cl.ly/qYr1" target="_blank"><img src="https://d3vv6lp55qjaqc.cloudfront.net/items/1e1j1S2b111m0Y100D0z/Screen%20Recording%202018-03-29%20at%2011.40%20PM.gif" style="display: block;height: auto;width: 100%;"/></a>
</details>
<details>
<summary>Admin order refund amount and line item</summary>
<img width="188" alt="screen shot 2018-03-30 at 12 38 48 am" src="https://user-images.githubusercontent.com/1883584/38127166-eb6445bc-33c2-11e8-9bf4-d11ee7307295.png">
<img width="966" alt="screen shot 2018-03-30 at 12 46 57 am" src="https://user-images.githubusercontent.com/1883584/38127173-f42c9f82-33c2-11e8-8ff1-e4e4256f3603.png">
</details>
<details>
<summary>Admin order delete tax</summary>
<a href="https://cl.ly/qYcr" target="_blank"><img src="https://d3vv6lp55qjaqc.cloudfront.net/items/0h0f3F120a132G0B340H/Screen%20Recording%202018-03-30%20at%2001.45%20AM.gif" style="display: block;height: auto;width: 100%;"/></a>
</details>
<details>
<summary>Admin order actions</summary>
<img width="343" alt="screen shot 2018-03-30 at 2 07 00 am" src="https://user-images.githubusercontent.com/1883584/38127210-1bf787fc-33c3-11e8-98d5-065a386fa088.png">
</details>
</details>

willgorham committed:
Import Gridicons library and files

willgorham committed:
Replace icon: Admin menu 'Products' item

willgorham committed:
Replace icon: Admin menu 'Sales reports' item

willgorham committed:
Change default fill color to gray to prevent initial flash

willgorham committed:
Add icon: Admin menu top-level 'Coupons' item

willgorham committed:
Compile CSS

willgorham committed:
Replace icon: Tooltip

willgorham committed:
Tabs not spaces

willgorham committed:
Replace icon: Checkmark on status page

willgorham committed:
Replace icon: Warning on status page

willgorham committed:
Replace icon: 'X' (cross) on status page

Tweaked CSS to account for svgs not inside mark element.

willgorham committed:
Update system status generator for new icons

willgorham committed:
Remove products list table image column header icon

willgorham committed:
Replace icon: Featured star in products list table

willgorham committed:
Replace icons: All product and coupon data meta box tabs

willgorham committed:
Replace icon: Coupons blank slate

willgorham committed:
Replace icon: Orders blank slate

willgorham committed:
Replace icon: Products blank slate

willgorham committed:
Replace icon: API Keys blank state

willgorham committed:
Replace icon: Webhooks blank state

willgorham committed:
Replace icons: Product category/product attirbute table reorder handles

willgorham committed:
Replace icon: Settings shipping zones globe

willgorham committed:
Spaces -> tabs

willgorham committed:
Merge branch 'master' into update/gridicons

willgorham committed:
Replace icons: Settings shipping zones + shipping zone methods reorder handle

willgorham committed:
Replace icon: Settings payment gateways reorder handle

willgorham committed:
Replace icons: Payment gateways + Emails enabled/disabled/manually sent

willgorham committed:
Remove icon: Settings emails configure cog

willgorham committed:
Replace icon: Reports export button

willgorham committed:
Replace icons: Order edit address + edit/delete item

willgorham committed:
Replace icons: Order shipping/fee/refund line item

willgorham committed:
Remove icon: Order item image placeholder

willgorham committed:
Replace order actions icon with text

willgorham committed:
Replace icon: Order preview

willgorham committed:
Replace icon: rating star

Includes:
- Single product rating
- Single product review rating
- Single product 'leave a review'
- Product loop
- Product sidebar
- Filter by rating widget

willgorham committed:
Tweak rating star styles

willgorham committed:
Replace icon: Shipping zones blank state globe

willgorham committed:
Remove icon: shipping notice globe

willgorham committed:
Replace icon: BACS table reorder handle

willgorham committed:
Replace icon: product downloads + variations sort handle

willgorham committed:
Replace icons: product variation image add/delete

willgorham committed:
Fix product Variations data tab icon

willgorham committed:
Clean up

willgorham committed:
Replace icons: AJAX add to cart loading/added

willgorham committed:
Replace icons: frontend message/info/notice/error

willgorham committed:
More tabs. Tabs are great...

willgorham commented:
Jeez, sorry about the mess of screenshots 🤦‍♂️ 
Originally I had these tied to individual commits, but in my personal repo. What's the best way to organize these for review? Happy to clean things up... 😬

willgorham commented on includes/class-wc-post-types.php:
```diff
@@ -312,6 +312,7 @@ public static function register_post_types() {
 					'has_archive'         => $has_archive,
 					'show_in_nav_menus'   => true,
 					'show_in_rest'        => true,
+					'menu_icon'           => 'data:image/svg+xml;base64,' . base64_encode( str_replace( '<svg', '<svg fill="black"', get_gridicon( 'gridicons-product' ) ) ), // SVG 'fill' attribute must be set for menu colors to apply correctly.
```
The `svg` tag here needs to have a defined `fill` attribute to get the admin menu color scheme to apply correctly. I could edit the svg directly, but the changes would get overwritten if the Gridicons library got updated. The `str_replace` isn't elegant, but it does the job.

willgorham replied:
The `str_replace` here doesn't feel great, but the icon has to have a defined `fill` attribute for the admin menu default colors to apply correctly. It's not pretty, but this does the job.

willgorham commented:
Replacing the metabox data tab icons here https://github.com/woocommerce/woocommerce/pull/19445/commits/23b0623aa2fd17afa54810c78ef315a55c2d052c is going to break some backwards compatibility with extensions that define their own tab icons (e.g. Product Bundles). But they can add a Gridicon name when defining a custom tab via `woocommerce_coupon_data_tabs` or `woocommerce_product_data_tabs`. Could go back to pseudo-element styling for compatibility if desired.

willgorham commented:
https://github.com/woocommerce/woocommerce/pull/19445/commits/a6bb9047fefccf106bf969eae36c9abd1b0a3afa affects WC Subscriptions adding their Automatic Recurring Payments column to the payment gateway table (not significantly, they just lose the 'enabled' icon). Could go back to pseudo-elements to make this work for others relying on this styling as well, though they could update the column content on their end pretty easily. 

<img width="1019" alt="screen shot 2018-03-15 at 8 49 40 pm" src="https://user-images.githubusercontent.com/1883584/37498065-71b7fcd0-2892-11e8-99eb-4f545c8a4ad0.png">

willgorham committed:
Replace icon: Order coupon remove X

willgorham committed:
Clean up

willgorham committed:
Replace icons: My account order details phone/mail

willgorham committed:
Replace icon: Rating filter widget remove X

jameskoster commented:
> The main Woo icon in the admin menu per #16903 (comment) (maybe a shopping cart?).

For now I think we should leave this. I can provide an svg if required.

> The product list table types column (pending whatever happens to that). There are Gridicons for simple, virtual, external, and downloadable products. We'd need something for variable and grouped.

There's a vote taking place over here https://woocommerce.wordpress.com/2018/02/28/what-should-happen-to-the-product-type-column-on-the-admin-products-screen/

> The order table actions buttons. I think we could just go with text here? I think I recall a past issue about these being confusing.

Now that there's more room in the orders table we can comfortably make these text buttons.

> Review star: This will cause styling breakage for themes that define their own star CSS

We have a separate font for the star. I think we should consider leaving that as-is for now and investigate an alternative solution for things like the rating selection in the review form in a separate PR.

> Front-end notices / Archive add to cart button ajax icons: Could affect custom themes. The button icons are pseudo-elemens also, so color customization is a challenge.

They were pseudo elements before, right? So it should affect themes. Custom themes should be dequeuing the WC css anyway.

> Product/coupon metabox data tabs: Will affect extensions that add their own icons. I added support for declaring a Gridicon when adding a custom tab.

Yeah this is tricky. Ideally extensions should also be able to specify a custom .svg file for the tab as Gridicons will not cover everything.

willgorham commented:
> For now I think we should leave this. I can provide an svg if required.

No need for an svg right away since the WooCommerce font is staying available for now, right? I'm happy to replace it though if you want to provide one.

> Now that there's more room in the orders table we can comfortably make these text buttons.

Sounds good - I'll take care of this.

> We have a separate font for the star. I think we should consider leaving that as-is for now

Makes sense, I'll revert the star changes.

> They were pseudo elements before, right? So it shouldn't affect themes.

Add-to-cart buttons should be fine because those are still pseudo-elements. The notice icons are inline now though, so those could be affected.

> Yeah this is tricky. Ideally extensions should also be able to specify a custom .svg file for the tab as Gridicons will not cover everything.

What do you think makes the most sense here, allowing an extension to overwrite the entire icon markup? (If that's done, then an extension could potentially add whatever image type they want, which I guess would be fine. Opens things up to more potential breakage, but I guess extensions already have plenty of those opportunities already 😄)

jameskoster commented:
> No need for an svg right away since the WooCommerce font is staying available for now, right?

True, we can leave for now.

> If that's done, then an extension could potentially add whatever image type they want

Yeah that's actually a good point. A mismatching icon style will negatively affect the overall UX. OK, I retract the custom icon requirement :) Dev's should be able to;

1. Do nothing and 'inherit' a default, generic tab icon
2. Select a Gridicon that fits their needs

If there's nothing currently in Gridicons suitable to their needs, we should encourage them to open an issue/PR on the Gridicons repo.

willgorham commented:
Cool, makes sense to me!

I guess theoretically an extension author could hide the svg and add a pseudo-element icon with whatever they want, if they really want to go that far. (Dynamic Pricing is already kind of doing that.) But keeping it to Gridicons with a default fallback and the opportunity to request new Gridicons seems reasonable.

willgorham committed:
Replace icon: Layered nav filters remove X

willgorham committed:
Replace icon: Products import complete checkmark

willgorham committed:
Replace icon: Reports chart widget resize

willgorham committed:
Replace icon: Product downloadable files delete X

willgorham committed:
Replace icon: Admin product gallery image remove X

willgorham committed:
Replace icon: Admin product gallery image placeholder

willgorham committed:
Replace icon: Admin order refund amount. Tweak refund line icon.

willgorham committed:
Replace icon: Admin order items delete tax X

willgorham committed:
Revert "Tweak rating star styles"

This reverts commit a4ff38db98f4dfb9381dae43886ede24a1325c68.

willgorham committed:
Revert "Replace icon: rating star"

This reverts commit 11668af15a3de44daf097d7a1cd8c7cbf523a9ff.

willgorham committed:
Convert admin order actions buttons to text-only

willgorham committed:
Fix unnecessary translation

willgorham committed:
Fix indentation

willgorham commented:
- Reverted the star rating changes. (Maybe something to note - the 'star' font is used everywhere except the Add A Review stars, where the 'WooCommerce' font is used.)

- Updated the order action buttons to be plain text.

- I also updated some icons that I missed on the first pass:

<details>
<summary>My Account email/phone</summary>
<img width="390" alt="screen shot 2018-03-18 at 5 04 38 pm" src="https://user-images.githubusercontent.com/1883584/38126949-c83d5ef8-33c1-11e8-9432-4ad6d3a3729d.png">
</details>
<details>
<summary>Layered nav, rating filter, and active filters widgets 'remove' X</summary>
<img width="365" alt="screen shot 2018-03-29 at 8 25 11 pm" src="https://user-images.githubusercontent.com/1883584/38127028-3d597bd6-33c2-11e8-964f-3faf9c5cd669.png">
</details>
<details>
<summary>Product importer complete check</summary>
<img width="723" alt="screen shot 2018-03-29 at 8 47 24 pm" src="https://user-images.githubusercontent.com/1883584/38127078-7eb0e72c-33c2-11e8-89af-76a3680c3490.png">
</details>
<details>
<summary>Reports - chart widget resize handle</summary>
<img width="250" alt="screen shot 2018-03-29 at 9 03 55 pm" src="https://user-images.githubusercontent.com/1883584/38127095-9a477bfe-33c2-11e8-9de7-cf01bec2d022.png">
</details>
<details>
<summary>Product downloadable files 'remove' X</summary>
<img width="549" alt="screen shot 2018-03-29 at 9 19 56 pm" src="https://user-images.githubusercontent.com/1883584/38127125-b5479c86-33c2-11e8-8c0b-7cf1fb0eff0b.png">
</details>
<details>
<summary>Admin product gallery remove X and placeholder</summary>
<a href="https://cl.ly/qYr1" target="_blank"><img src="https://d3vv6lp55qjaqc.cloudfront.net/items/1e1j1S2b111m0Y100D0z/Screen%20Recording%202018-03-29%20at%2011.40%20PM.gif" style="display: block;height: auto;width: 100%;"/></a>
</details>
<details>
<summary>Admin order refund amount and line item</summary>
<img width="188" alt="screen shot 2018-03-30 at 12 38 48 am" src="https://user-images.githubusercontent.com/1883584/38127166-eb6445bc-33c2-11e8-9bf4-d11ee7307295.png">
<img width="966" alt="screen shot 2018-03-30 at 12 46 57 am" src="https://user-images.githubusercontent.com/1883584/38127173-f42c9f82-33c2-11e8-8ff1-e4e4256f3603.png">
</details>
<details>
<summary>Admin order delete tax</summary>
<a href="https://cl.ly/qYcr" target="_blank"><img src="https://d3vv6lp55qjaqc.cloudfront.net/items/0h0f3F120a132G0B340H/Screen%20Recording%202018-03-30%20at%2001.45%20AM.gif" style="display: block;height: auto;width: 100%;"/></a>
</details>
<details>
<summary>Admin order actions</summary>
<img width="343" alt="screen shot 2018-03-30 at 2 07 00 am" src="https://user-images.githubusercontent.com/1883584/38127210-1bf787fc-33c3-11e8-98d5-065a386fa088.png">
</details>
I've added these items to the main issue comment as well.

willgorham committed:
Resolve merge conflicts

willgorham committed:
Use wpspin-2x so loader looks better

willgorham committed:
Tweak order line refund item spacing

willgorham commented:
Hi @mikejolley 

Sorry about the delay here, just got back from vacation. 

I totally understand about the size of this PR, half way through I realized it was going to be a lot bigger than I initially thought 😬. Figured I'd just submit it and break it up if needed - so I'll work on splitting this into more reasonable chunks. 

Thanks!

willgorham commented:
Is this still relevant with wc-admin coming down the pipe? If it is, I'm happy to go ahead and break this up into smaller chunks for review when I get the chance.

mikejolley commented:
@willgorham wc-admin (the dashboard part) will likely ship before 4.0. Perhaps it's best to leave this for now and revisit in the future since we're unsure how long this change would actually be relevent. There are other places that require more immediate action :) Thanks!

willgorham commented:
@mikejolley Sounds good to me - thanks!
